ursula_cli/shell.py

```diff
@@ -278,20 +285,28 @@ def _run_heat(args, hot):
         else:
             raise Exception(e)
 
-    if not stack_exists:
+    stack_action = 'create'
+
+    if stack_exists and args.heat_stack_update:
+        stack_action = 'update'
+        LOG.debug("Updating stack")
+        heatclient.stacks.update(stack_name, **STACK)
+        time.sleep(5)
+    elif not stack_exists:
         LOG.debug("Creating stack")
         heatclient.stacks.create(**STACK)
         time.sleep(5)
-        while heatclient.stacks.get(stack_name).status == 'IN_PROGRESS':
-            LOG.debug("Waiting on stack creation...")
-            time.sleep(5)
+
+    while heatclient.stacks.get(stack_name).status == 'IN_PROGRESS':
+        LOG.debug("Waiting on stack...")
+        time.sleep(5)
```

Comment:
Is it possible that this while loop could last as long as there's an `IN_PROGRESS` stack creation (not forever, but almost)?  Maybe we want a time limit in there?